zebra-chain/src/parallel/tree.rs

```diff
@@ -49,32 +56,18 @@ impl NoteCommitmentTrees {
         &mut self,
         block: &Arc<Block>,
     ) -> Result<(), NoteCommitmentTreeError> {
-        self.update_trees_parallel_list(
-            [(
-                block
-                    .coinbase_height()
-                    .expect("height was already validated"),
-                block.clone(),
-            )]
-            .into_iter()
-            .collect(),
-        )
-    }
+        let block = block.clone();
+        let height = block
+            .coinbase_height()
+            .expect("height was already validated");
+        let block_list: BTreeMap<Height, Arc<Block>> = [(height, block)].into_iter().collect();
```

Comment:
Optional: this isn't a block list any more, so we don't need to fake a one-item list.

With one block, this code:
```rust
        block_list
            .values()
            .flat_map(|block| block.transactions.iter())
```

Is the same as this code:
```rust
        block.transactions.iter()
```